src/bin/wasmtime.rs

```diff
@@ -5,6 +5,8 @@
 
 use anyhow::Result;
 use clap::Parser;
+#[cfg(all(feature = "component-model", feature = "wasi-http"))]
```

Comment:
Another possibility here, if you're interested,  is to model this as subcommands-as-features. The `wasm-tools` CLI does this where there's a Cargo feature for each subcommand and they're all enabled by default. That way this could be enabled with `cargo build --features serve` and `serve` would implicitly enable the component model and `wasi-http`

Reply:
I'll leave the decision up to y'all though, I don't feel strongly either way

Reply:
I like that!

Reply:
Actually, `wasi-http` already enables `component-model`, so we could make this only depend on `wasi-http`. However, I don't like that as much because it's not quite as declarative as the `serve` command feature.